source/module_io/write_Vxc.hpp

```diff
@@ -149,6 +149,25 @@ namespace ModuleIO
         return e;
     }
 
+    template<typename T>
+    std::vector<double> orbital_energy(
+        const int ik,
+        const int nbands,
+        const std::vector<T>& mat_mo,
+        const Parallel_2D& p2d)
+    {
+        std::vector<double> e(nbands, 0.0);
+        for (int i = 0;i < nbands;++i)
+        {
+            if (p2d.in_this_processor(i, i))
+            {
+                e[i] = get_real(mat_mo[p2d.global2local_col(i) * p2d.get_row_size() + p2d.global2local_row(i)]);
```

Comment:
I would suggest using a variable to record the index = p2d.global2local_col(i) * p2d.get_row_size() + p2d.global2local_row(i)], and then use mat_mo[index]

Reply:
I've changed it according to your suggestion. But Why? Here "*" and "+" calculations are not repeated so `index` is not reused.